### DISCUSSION TITLE
config questions

### DISCUSSION THREAD

**Discussion by meelten** [anatol/booster#216] [General]
Hello, I have two easy questions about configuration file:

1. Does order of flags in booster.yaml matter in any case or can be completely random?

2. Do I assume correctly that for boolean flags default is always false? Manual says it specifically only about "universal" flag, but for say "strip" I'm not sure whether it's on or off by default. If I don't use lvm or vconsole should I add it to config and set to false?

**Comment by anatol** [comment]
Hello @meelten 

Parsing the booster config file is done by `yaml.Unmarshal` function here https://github.com/anatol/booster/blob/8de6af58c11db7c4370ee8df6d38d0b7aeca0cd1/generator/config.go#L54

And answers to your questions come from the `Unmarshal` function implementation.

> Does order of flags in booster.yaml matter in any case or can be completely random?

The order of the fields in the yaml file does not matter.

> Do I assume correctly that for boolean flags default is always false?

The defaults for boolean flags are `false` indeed. Some values are overwritten by command line flags like `--universal` and `--strip`, see `https://github.com/anatol/booster/blob/8de6af58c11db7c4370ee8df6d38d0b7aeca0cd1/generator/config.go#L145

Otherwise the flags are off by default.

> If I don't use lvm or vconsole should I add it to config and set to false?

No, you don't need these `false` flags to be added to the config.

**Reply by meelten** [reply]
Thank you very much for detailed explanation.
